### ISSUE TITLE
Getting "Cannot read property 'initialize' of undefined"

### ISSUE THREAD

**Issue by BigPun86** [gcrabtree/react-native-socketio#7]
![screen shot 2016-06-29 at 12 23 29](https://cloud.githubusercontent.com/assets/7880624/16448924/7e2c06e4-3df4-11e6-942a-303494d5d447.png)

![screen shot 2016-06-29 at 12 25 04](https://cloud.githubusercontent.com/assets/7880624/16448930/87ad56d2-3df4-11e6-8da1-0c781d825843.png)

<img width="509" alt="screen shot 2016-06-29 at 12 25 16" src="https://cloud.githubusercontent.com/assets/7880624/16448973/c4455b6c-3df4-11e6-98f6-d0be61750007.png">

I notified when integrating this module to my xcode project, it gives me some warnings:

<img width="1115" alt="screen shot 2016-06-29 at 12 27 22" src="https://cloud.githubusercontent.com/assets/7880624/16449004/f20f3e64-3df4-11e6-81da-5a3f44aa5215.png">

Did anyone experience similar issue?

Running on RN 28 and testing with your example code

**Comment by mcliquid** [comment]
+1

**Comment by gcrabtree** [comment]
I'll take a look sometime tonight. 

I haven't upgraded support to RN 0.28 as the project we are working on had major issues with that release (outside of this library). We are currently running on RN 0.25.1.

Will keep you posted.

**Comment by gcrabtree** [comment]
Yeah, there are breaking API changes between those two releases (RN 0.25 and RN 0.28).
That would explain the deprecation warnings. I will migrate those to the new RCTEventEmitter class in a new release that supports 28+. They should probably still work in RN 28, but I'll need to update them moving forward.

@BigPun86 or @mcliquid  WRT the undefined error, how are you importing socketio?
I need to re-read the documentation that I have. The previous project (that appears to be dead) that I forked this from (spiritually, not an actual fork), had that original ios example. It's likely not working so I'll need to fix that up as well.

For now, this is how I include socketio and have it working on RN 25:

```
let io = require('react-native-socketio');
let socket = new io(url, socketioOptions); 
socket.connect();
```

Let me know if that gets you past the undefined problem.

There are a few issues with the packager with RN 28.
https://github.com/facebook/react-native/issues/8297
https://github.com/facebook/react-native/issues/8409

**Comment by GeoffreyPlitt** [comment]
I can confirm this is broken with the latest RN (0.32). Does anyone have a fix?

In the meantime, I'd suggest a big warning on the README so people don't waste their time.

**Comment by jitenderchand1** [comment]
I am getting the this.socket undefined with react native 0.35

**Comment by i25878427** [comment]
I am getting the this.socket undefined with react native 0.31

**Comment by huangjilaiqin** [comment]
I am getting the this.socket undefined with react native 0.37.0

**Comment by huangjilaiqin** [comment]
I am getting the this.socket undefined with react native 0.37.0

Answer：
I forgot change getPackages on MainApplication.java

README.md 3.Register module (in MainActivity.java)    is error!!!
is should change to:  3.Register module (in MainApplication.java)
@gcrabtree hope help

**Comment by LucasSouzaa** [comment]
i'm getting this issue in react-native 0.42 in iOS only

**Comment by Darex1991** [comment]
The same. Error on iOS 
I tried 0.39.2, 0.40.0, 0.41.0, 0.42.3 and this bug still exist.

> TypeError: Cannot read property 'initialize' of undefined
>     at new Socket (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:68738:13)
>     at GameLobby.createLobbySocket (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:69173:12)
>     at GameLobby.componentWillMount (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:69040:37)
>     at http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:19500:24
>     at measureLifeCyclePerf (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:19117:8)
>     at ReactCompositeComponentWrapper.performInitialMount (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:19499:1)
>     at ReactCompositeComponentWrapper.mountComponent (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:19378:13)
>     at Object.mountComponent (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:17462:29)
>     at Object.obj.(anonymous function) [as mountComponent] (eval at initialize (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:34422:1), <anonymous>:1217:27)
>     at Constructor.mountChildren (http://localhost:8081/index.ios.bundle?platform=ios&dev=true&minify=false:18389:32)

**Comment by atlas1119** [comment]
who can fix this problem???

**Comment by felansu** [comment]
+1

**Comment by gogulanareshkw** [comment]
any update on this issue???

**Comment by felansu** [comment]
@gogulanareshkw no.

**Comment by ciriac** [comment]
Is there really a need to still be using this library? If you're running the latest RN version you can just use socket.io-client without any issue.

`yarn add socket.io-client`

```
import SocketIOClient from 'socket.io-client';

const socket = new SocketIOClient('your_domain', {
   transports: ['websocket']
});

socket.connect();

socket.on('connect', () => {
   console.log('connected!');
});
```
